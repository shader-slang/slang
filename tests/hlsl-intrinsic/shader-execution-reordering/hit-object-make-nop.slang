// hit-object-make-nop.slang

//TEST:SIMPLE: -target dxil -entry rayGenerationMain -stage raygeneration -profile sm_6_5 -DNV_SHADER_EXTN_SLOT=u0 

//DISABLE_TEST(compute):COMPARE_COMPUTE:-d3d12 -output-using-type -use-dxil -profile sm_6_5 -render-feature ray-query
//DISABLE_TEST(compute):COMPARE_COMPUTE:-vk -output-using-type -render-feature ray-query

//TEST_INPUT:set outputBuffer = out ubuffer(data=[0, 0, 0, 0], stride=4)
RWStructuredBuffer<uint> outputBuffer;

void rayGenerationMain() 
{
    int2 launchID = int2(DispatchRaysIndex().xy);
    int2 launchSize = int2(DispatchRaysDimensions().xy);

    int idx = launchID.x;

    HitObject hit = HitObject::MakeNop();
    
    outputBuffer[idx] = uint(hit.IsNop());
}
