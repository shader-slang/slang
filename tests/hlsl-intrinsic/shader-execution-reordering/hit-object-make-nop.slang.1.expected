result code = 0
standard error = {
}
standard output = {
#version 460
#extension GL_EXT_ray_tracing : require
#extension GL_NV_shader_invocation_reorder : require
layout(row_major) uniform;
layout(row_major) buffer;
layout(std430, binding = 0) buffer _S1 {
    uint _data[];
} outputBuffer_0;
void main()
{
    uvec3 _S2 = ((gl_LaunchIDEXT));
    ivec2 launchID_0 = ivec2(_S2.xy);
    uvec3 _S3 = ((gl_LaunchSizeEXT));
    int idx_0 = launchID_0.x;
    hitObjectNV hitObj_0;
    hitObjectRecordEmptyNV((hitObj_0));
    uint _S4 = uint(idx_0);
    bool _S5 = (hitObjectIsEmptyNV((hitObj_0)));
    uint _S6 = uint(_S5);
    ((outputBuffer_0)._data[(_S4)]) = _S6;
    return;
}

}
